File: nl.naturalis.nba.client.php/model/VernacularName.php
<?php

namespace nl\naturalis\nba\client\php\model;

use nl\naturalis\nba\client\php\model\NBADomainObject;
use nl\naturalis\nba\client\php\helper\ReferenceList;
use nl\naturalis\nba\client\php\helper\ExpertList;

class VernacularName extends NBADomainObject {

	private $name;
	private $language;
	private $preferred;
	private $references;
	private $experts;

	public function __construct(string $name)
	{
		$this->name = $name;
	}

	public function getName()
	{
		return $this->name;
	}

	public function setName(string $name)
	{
		$this->name = $name;
	}

	public function getLanguage()
	{
		return $this->language;
	}

	public function setLanguage(string $language)
	{
		$this->language = $language;
	}

	public function getPreferred()
	{
		return $this->preferred;
	}

	public function setPreferred(bool $preferred)
	{
		$this->preferred = $preferred;
	}

	/**
	 *
	 * @return \nl\naturalis\nba\client\php\helper\ReferenceList
	 */
	public function getReferences()
	{
		return $this->references;
	}

	public function setReferences(ReferenceList $references)
	{
		$this->references = $references;
	}

	/**
	 *
	 * @return \nl\naturalis\nba\client\php\helper\ExpertList
	 */
	public function getExperts()
	{
		return $this->experts;
	}

	public function setExperts(ExpertList $experts)
	{
		$this->experts = $experts;
	}
}

File: nl.naturalis.nba.api/src/main/java/nl/naturalis/nba/api/model/VernacularName.java
package nl.naturalis.nba.api.model;

import static nl.naturalis.nba.api.annotations.Analyzer.CASE_INSENSITIVE;
import static nl.naturalis.nba.api.annotations.Analyzer.DEFAULT;
import static nl.naturalis.nba.api.annotations.Analyzer.LIKE;

import java.util.List;

import nl.naturalis.nba.api.annotations.Analyzers;

public class VernacularName implements INbaModelObject {

	@Analyzers({ DEFAULT, CASE_INSENSITIVE, LIKE })
	private String name;
	private String language;
	private Boolean preferred;

	private List<Reference> references;
	private List<Expert> experts;

	public VernacularName()
	{
	}

	public VernacularName(String name)
	{
		this.name = name;
	}

	public String getName()
	{
		return name;
	}

	public void setName(String name)
	{
		this.name = name;
	}

	public String getLanguage()
	{
		return language;
	}

	public void setLanguage(String language)
	{
		this.language = language;
	}

	public Boolean getPreferred()
	{
		return preferred;
	}

	public void setPreferred(Boolean preferred)
	{
		this.preferred = preferred;
	}

	public List<Reference> getReferences()
	{
		return references;
	}

	public void setReferences(List<Reference> references)
	{
		this.references = references;
	}

	public List<Expert> getExperts()
	{
		return experts;
	}

	public void setExperts(List<Expert> experts)
	{
		this.experts = experts;
	}

	@Override
	public boolean equals(Object obj)
	{
		if (this == obj) {
			return true;
		}
		if (obj == null || !(obj instanceof VernacularName)) {
			return false;
		}
		VernacularName other = (VernacularName) obj;
		return eq(name, other.name) && eq(language, other.language);
	}

	@Override
	public int hashCode()
	{
		int hash = 17;
		hash = (hash * 31) + name.hashCode();
		hash = (hash * 31) + language == null ? 0 : language.hashCode();
		return hash;
	}

	@Override
	public String toString()
	{
		StringBuilder sb = new StringBuilder(64);
		sb.append(name);
		if (language != null) {
			sb.append(" (").append(language).append(')');
		}
		return sb.toString();
	}

	private static boolean eq(Object obj0, Object obj1)
	{
		if (obj0 == null) {
			if (obj1 == null)
				return true;
			return false;
		}
		return obj1 == null ? false : obj0.equals(obj1);
	}
}
